exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonNonNumerics.java

```diff
@@ -0,0 +1,167 @@
+    File file = new File(getTempDir("nan_test"), "nan_test.json");
+    String json = "{\"nan\":NaN, \"inf\":Infinity}";
+    String query = String.format("select * from dfs.`%s`",file.getAbsolutePath());
+    try {
+      FileUtils.writeStringToFile(file, json);
+      test("alter session set `store.json.reader.non_numeric_numbers` = true");
+      testBuilder()
+        .sqlQuery(query)
+        .unOrdered()
+        .baselineColumns("nan", "inf")
+        .baselineValues(Double.NaN, Double.POSITIVE_INFINITY)
+        .build()
+        .run();
+    } finally {
+      test("alter session reset `store.json.reader.non_numeric_numbers`");
+      FileUtils.deleteQuietly(file);
+    }
+  }
+
+  @Test(expected = UserRemoteException.class)
+  public void testNonNumericFailure() throws Exception {
+    File file = new File(getTempDir("nan_test"), "nan_test.json");
+    test("alter session set `store.json.reader.non_numeric_numbers` = false");
+    String json = "{\"nan\":NaN, \"inf\":Infinity}";
+    try {
+      FileUtils.writeStringToFile(file, json);
+      test("select * from dfs.`%s`;", file.getAbsolutePath());
+    } catch (UserRemoteException e) {
+      assertThat(e.getMessage(), containsString("Error parsing JSON"));
+      throw e;
+    } finally {
+      test("alter session reset `store.json.reader.non_numeric_numbers`");
+      FileUtils.deleteQuietly(file);
+    }
+  }
+
+  @Test
+  public void testCreateTableNonNumerics() throws Exception {
+    File file = new File(getTempDir("nan_test"), "nan_test.json");
+    String json = "{\"nan\":NaN, \"inf\":Infinity}";
+    String tableName = "ctas_test";
+    try {
+      FileUtils.writeStringToFile(file, json);
+      test("alter session set `store.json.reader.non_numeric_numbers` = true");
+      test("alter session set `store.json.writer.non_numeric_numbers` = true");
+      test("alter session set `store.format`='json'");
+      test("create table dfs_test.tmp.`%s` as select * from dfs.`%s`;", tableName, file.getAbsolutePath());
+
+      // ensuring that `NaN` and `Infinity` tokens ARE NOT enclosed with double quotes
+      File resultFile = new File(new File(getDfsTestTmpSchemaLocation(),tableName),"0_0_0.json");
+      String resultJson = FileUtils.readFileToString(resultFile);
+      int nanIndex = resultJson.indexOf("NaN");
+      assertFalse("`NaN` must not be enclosed with \"\" ", resultJson.charAt(nanIndex - 1) == '"');
+      assertFalse("`NaN` must not be enclosed with \"\" ", resultJson.charAt(nanIndex + "NaN".length()) == '"');
+      int infIndex = resultJson.indexOf("Infinity");
+      assertFalse("`Infinity` must not be enclosed with \"\" ", resultJson.charAt(infIndex - 1) == '"');
+      assertFalse("`Infinity` must not be enclosed with \"\" ", resultJson.charAt(infIndex + "Infinity".length()) == '"');
+    } finally {
+      test("drop table if exists dfs_test.tmp.`%s`", tableName);
+      test("alter session reset `store.json.reader.non_numeric_numbers`");
```

Comment:
Should be done in a finally block to ensure these get reset even if the test fails.